src/token.ts

```diff
@@ -74,6 +78,80 @@ export async function fetchAppToken(
     };
 }
 
+/**
+ * Crowdin Agent Authentication
+ */
+
+interface FetchAgentTokenArgs extends FetchAppTokenArgs {
+    agentId?: number;
+}
+
+function isFetchAgentTokenArgs(args: FetchAgentTokenArgs | string): args is FetchAgentTokenArgs {
+    return typeof args !== 'string';
+}
+
+/**
+ * @param appId Crowdin app identifier
+ * @param appSecret Crowdin app secret received from install event
+ * @param clientId OAuth client id of the app
+ * @param clientSecret OAuth client secret of the app
+ * @param domain Crowdin organization domain
+ * @param userId The user who installed the application
+ * @param agentId The agent who will perform the actions
+ * @param url Custom url for token exchange
+ * @returns token object which is needed to establish communication between app and Crowdin API
+ */
+export async function fetchAgentToken(
+    appId: string,
+    appSecret: string,
+    clientId: string,
+    clientSecret: string,
+    domain: string,
+    userId: number,
+    agentId?: number,
+    url?: string,
+): Promise<AppToken>;
+
+export async function fetchAgentToken(args: FetchAgentTokenArgs): Promise<AppToken>;
+
+export async function fetchAgentToken(
+    appIdOrArgs: string | FetchAgentTokenArgs,
+    appSecret?: string,
+    clientId?: string,
+    clientSecret?: string,
+    domain?: string,
+    userId?: number,
+    agentId?: number,
+    url?: string,
+): Promise<AppToken> {
+    let options: FetchAgentTokenArgs;
+    if (isFetchAgentTokenArgs(appIdOrArgs)) {
```

Comment:
I don't think we need two ways of passing args here. It's a new function, let's have only object-based arguments. Others have it for backward compatibility

Reply:
It makes sense. Fixed it